testhelpers/docker_registry.go

```diff
@@ -103,34 +159,85 @@ func (r *DockerRegistry) Start(t *testing.T) {
 	var authHeaders map[string]string
 	if r.username != "" {
 		// Write Docker config and configure auth headers
-		writeDockerConfig(t, r.DockerDirectory, r.Port, r.encodedAuth())
+		writeDockerConfig(t, r.DockerDirectory, r.Host, r.Port, r.encodedAuth())
 		authHeaders = map[string]string{"Authorization": "Basic " + r.encodedAuth()}
 	}
 
 	// Wait for registry to be ready
 	Eventually(t, func() bool {
-		txt, err := HTTPGetE(fmt.Sprintf("http://localhost:%s/v2/_catalog", r.Port), authHeaders)
+		txt, err := HTTPGetE(fmt.Sprintf("http://%s:%s/v2/_catalog", r.Host, r.Port), authHeaders)
 		return err == nil && txt != ""
 	}, 100*time.Millisecond, 10*time.Second)
 }
 
 func (r *DockerRegistry) Stop(t *testing.T) {
-	t.Log("stop registry")
 	t.Helper()
+	t.Log("stop registry")
+
 	if r.Name != "" {
 		DockerCli(t).ContainerKill(context.Background(), r.Name, "SIGKILL")
 		DockerCli(t).ContainerRemove(context.TODO(), r.Name, types.ContainerRemoveOptions{Force: true})
 	}
+
+	if r.volumeName != "" {
+		// try to cleanup shared volume if this is the last user
+		err := DockerCli(t).VolumeRemove(context.Background(), r.volumeName, false)
+		if err != nil {
+			// fail if err is not from volume in use
+			if !errdefs.IsConflict(err) {
+				AssertNil(t, err)
+			}
+		}
+	}
 }
 
 func (r *DockerRegistry) RepoName(name string) string {
-	return "localhost:" + r.Port + "/" + name
+	return r.Host + ":" + r.Port + "/" + name
 }
 
 func (r *DockerRegistry) EncodedLabeledAuth() string {
 	return base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf(`{"username":"%s","password":"%s"}`, r.username, r.password)))
 }
 
+func DockerHostname(t *testing.T) string {
```

Comment:
Would it be worth adding a docs comment here describing the chain of precedence?